badge_server/test_badges.py

```diff
@@ -593,5 +626,358 @@ def test_github_py2py3_py3_incompatible_fresh_nodeps(self):
         self.assertEqual(json_response['left_text'],
                          'compatibility check (master)')
         self.assertEqual(json_response['right_text'], 'self incompatible')
+
+class TestBadgeImageOutdatedDependency(BadgeImageTestCase):
```

Comment:
It can be in another PR, but we also need to test the success case with dependencies.

Reply:
Good point - will do in a follow up PR

Reply:
Actually I think that this case will be covered once dependencies are added to the CRs in `RECENT_SUCCESS_DATA`

Reply:
I would imagine that the success case will have no have non-trivial dependencies - I would just add self dependencies to `RECENT_SUCCESS_DATA` since the data is not correct without it.